decidim-core/app/services/decidim/resource_search.rb

```diff
@@ -39,12 +39,22 @@ def search_category_id
     # `"global"` as parameter, and in the method we do the needed changes to search
     # properly.
     #
-    # You can use the `search_organization_scopes` helper method, defined in
-    # `Decidim::OrganizationScopesHelper`, to render the collection needed for the
+    # You can use the `search_participatory_process_subscopes` helper method, defined in
+    # `Decidim::ParticipatoryProcessScopesHelper`, to render the collection needed for the
     # `collection_check_boxes` form method.
     def search_scope_id
-      clean_scope_ids = [scope_id].flatten.map { |id| id == "global" ? nil : id }
-      query.where(decidim_scope_id: clean_scope_ids)
+      clean_scope_ids = [scope_id].flatten
+
+      conditions = []
+      if clean_scope_ids.delete("global")
+        conditions << "decidim_scope_id IS NULL"
+      end
+      if clean_scope_ids.any?
+        self.query = query.joins("LEFT JOIN decidim_scopes ON decidim_scope_id=decidim_scopes.id")
+        conditions.concat ["? = ANY(decidim_scopes.part_of)"] * clean_scope_ids.count
```

Comment:
This method receives a list of scopes ids to find, maybe including "global" keyword for global resources.
On one hand, you could have to retrieve all resources whose scope in part of any of the scopes on the list.
On the other hand, you could have to retrieve resources with no scope, that is global.
If the user ask for both conditions, sentences are joined by an OR, so I can't use an inner join because it will exclude resources with no scope.

I don't like this solution very much; the right one would require that every process has a scope. That would means that "Global" scope should be the parent scope for every top level scopes, because [nothing is something](https://www.youtube.com/watch?v=OMPfEXIlTVE). But I'm not sure if it raises too much the complexity.